service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/BankService.kt

```diff
@@ -72,25 +74,57 @@ class BankService(
         }
 
     fun redeemBurnCoin(uuid: UUID, amount: BigDecimal) =
-        synchronized(CoinRedeemBurnRecord::class.java) {
+        synchronized(TxRequestViewRecord::class.java) {
             transaction {
                 log.info("Redeem burning coin for $uuid for amount $amount")
-                check(CoinRedeemBurnRecord.findById(uuid) == null) { "Coin redeem burn request for uuid $uuid already exists" }
+                check(TxRequestViewRecord.findById(uuid) == null) { "Tx request for uuid $uuid already exists" }
 
                 val coinAmount = amount.toCoinAmount()
                 // Account for any pending records in progress by netting out from balance lookups
-                val pendingAmount = CoinRedeemBurnRecord.findPending()
-                    .fold(0L) { acc, record -> acc + record.coinAmount }
-                    .toBigInteger()
-
+                val pendingAmount = CoinRedeemBurnRecord.findPendingAmount()
                 val dccBalance = pbcService.getCoinBalance(pbcService.managerAddress, serviceProperties.dccDenom)
-                    .toBigInteger() - pendingAmount
+                    .toBigInteger() - pendingAmount - CoinTransferRecord.findPendingAmount()
                 check(coinAmount <= dccBalance) { "Insufficient dcc coin $dccBalance" }
 
+                // Make sure the bank has sufficient escrowed bank token
                 val markerEscrowBalance = pbcService.getMarkerEscrowBalance().toBigInteger() - pendingAmount
                 check(coinAmount <= markerEscrowBalance) { "Insufficient bank reserve coin escrowed $markerEscrowBalance" }
 
                 CoinRedeemBurnRecord.insert(uuid, amount)
             }
         }
+
+    fun transferCoin(uuid: UUID, bankAccountUuid: UUID?, blockchainAddress: String?, amount: BigDecimal) =
+        synchronized(TxRequestViewRecord::class.java) {
+            transaction {
+                log.info("Transferring coin for $uuid to bank account $bankAccountUuid or address $blockchainAddress for amount $amount")
+                check(TxRequestViewRecord.findById(uuid) == null) { "Tx request for uuid $uuid already exists" }
+
+                val coinAmount = amount.toCoinAmount()
+                // Account for any pending records in progress by netting out from balance lookups
+                val pendingAmount = CoinRedeemBurnRecord.findPendingAmount() + CoinTransferRecord.findPendingAmount()
+                val dccBalance = pbcService.getCoinBalance(denom = serviceProperties.dccDenom)
```

Comment:
nit - logic to account for sufficient balance is repeated twice

Reply:
there was enough of a variance between this and redeemBurnCoin - i didnt bother trying to DRY it